### PR TITLE
Fix stop kill multi notifications

### LINKED ISSUE #2618: stop --kill triggers failed event multiple times
First reported by @steoxley and easily repeatable using a simple suite.

```
[scheduling]
    [[dependencies]]
        graph = t1
[runtime]
    [[t1]]
        script = sleep 300
        [[[job]]]
            batch system = pbs
        [[[events]]]
            mail events = failed
```

`cylc run` the suite, wait for the task job to start. Run `cylc stop --kill` to get multiple emails.

It looks like a failed event is triggered multiple times when:
* The kill command is issued.
* The suite receives a message from the job to say that it has been terminated.
* The suite polls the job to confirm its termination.

I guess the suite should only emit a task failed event on actual status change.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Fix #2618.

`stop --kill`: task events should no longer trigger multiple times.
Incoming message confirm polling should now be done a bit more efficiently, and should no longer affect logic of non-incoming.

(This branch is currently based on the branch for #2593, so only the final 2 commits are relevant.)

### CODE REVIEW

File: lib/cylc/task_events_mgr.py [hunk 1]
```diff
@@ -346,43 +351,46 @@ def process_message(self, itask, severity, message, poll_func,
...
             self._process_message_submitted(itask, event_time)
         elif message.startswith(FAIL_MESSAGE_PREFIX):
             # Task received signal.
             signal = message[len(FAIL_MESSAGE_PREFIX):]
             self._db_events_insert(itask, "signaled", signal)
             self.suite_db_mgr.put_update_task_jobs(
                 itask, {"run_signal": signal})
-            if self._poll_to_confirm(itask, TASK_STATUS_FAILED, poll_func):
-                return
+            if (flag == self.INCOMING_FLAG
```
Comment (oliver-sanders) [inline lib/cylc/task_events_mgr.py]: Why does this line not come immediately after `elif message.startswith(FAIL_MESSAGE_PREFIX):`?

Comment (matthewrmshin) [reply lib/cylc/task_events_mgr.py]: Logic updated.

**Comment by matthewrmshin** [comment]
Yes, I have tried to design a test for this, but it is very difficult to get it to fail on master. It relies on a set up of PBS managing a large resource with many jobs to have the whole process slowed down enough to repeat this issue. It is very reproducible at our site at time, but become less so at other time. I give up at the end because I don't really want to add a test that can become a placebo.

**Comment by hjoliver** [comment]
Ok that's fine.

**Pull request merged**